xarray/tests/test_plot.py

```diff
@@ -1194,6 +1194,11 @@ def test_discrete_colormap_provided_boundary_norm(self) -> None:
         primitive = self.darray.plot.contourf(norm=norm)
         np.testing.assert_allclose(primitive.levels, norm.boundaries)
 
+    def test_discrete_colormap_provided_boundary_norm_matching_cmap_levels(self):
```

Comment:
Could you enable type checking by using `def test...(..) -> None:`

Probably you need to add some further asserts for optional attributes.

Reply:
type checking was enabled in the latest push. This test already succesfully tests the implementation (and fails when it is changed back), so i would say this covers it quite ok already. Do you have suggestions for other asserts?

Reply:
What I meant was that mypy sometimes complains if you try to access attributes of objects that might be None.
E.g. `x.colorbar.norm`: if colorbar is None, then accessing norm will raise an AttributeError.

But in this case it seems to work fine :)